audio-worklet/noisegate-test.html

```diff
@@ -0,0 +1,432 @@
+<!--
+Copyright 2017 Google Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<!DOCTYPE html>
+  <head>
+    <script src="test/testharness.js"></script>
+    <script src="test/testharnessreport.js"></script>
+    <script src="test/audit-util.js"></script>
+    <script src="test/audit.js"></script>
+    <script src="src/noisegate-script-processor.js"></script>
+
+    <script id="noisegate-test-code">
+      // Sample rate is a power of two to avoid interpolation from delay node.
+      // It is held artificially low for easy inspection of audio buffers.
+      const sampleRate = 4096;
```

Comment:
If I changed the sampleRate to 12345, will the test still pass?  AFAICT, there are now experimentally determined thresholds, so these tests should all pass if I change the sample rate.